.github/workflows/release_github.yml

```diff
@@ -39,119 +85,120 @@ jobs:
           data-directory: ${{ env.EIT_PROCESSING_TEST_DATA }}
       - name: Run pytest
         run: pytest -v
-      - name: Install eitprocessing with dev dependencies
+      - name: Install additional dependencies for building
         run: python3 -m pip install --upgrade ".[publishing]"
       - name: Build eitprocessing
         run: python3 -m build
-  
 
-  github_release:
+  merge_and_bump: 
+    name: Merge the changes into main and bump the version
     needs: test_python_releases
-    runs-on: "ubuntu-latest"
-    defaults:
-      run:
-        shell: bash -l {0}
-
+    runs-on: ubuntu-latest
+    outputs:
+      new-version: ${{ steps.bump.outputs.current-version }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
-      - name: Fail if main branch was selected
-        if: ${{ github.ref_name == 'main' }}
-        run: |
-          echo "Cannot release from main branch, please select valid release branch."
-          exit 1
-
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          token: ${{ secrets.GH_PAT }}
           ref: main
           fetch-depth: 0
-
       - name: Configure git
         run: |
-          git config user.email "actions@github.com"
+          git config user.email "${GITHUB_ACTOR_ID}+${GITHUB_ACTOR}@users.noreply.github.com"
           git config user.name "GitHub Actions"
-          git pull
+          git config -l
 
-      - name: Merge changes into main
+      - name: Merge branch into main
         run: |
           git switch main
-          git branch ${{ github.ref_name }} origin/${{ github.ref_name }} 
-          git merge ${{ github.ref_name }} --no-ff --no-commit
-          git commit --no-edit
-
-      - name: Bump version
+          git branch -f ${{ github.ref_name }} origin/${{ github.ref_name }} 
+          git merge ${{ github.ref_name }} --no-ff --no-edit
+    
+      - name: Install Python
+        uses: actions/setup-python@v5.1.1
+        with:
+          python-version: '3.12'
+  
+      - name: Install bump-my-version
+        shell: bash
+        run: pip install "bump-my-version==0.28.1"
```

Comment:
shall we fetch the version specified in pyproject, similar to how you did it for the [linting workflow](https://github.com/EIT-ALIVE/eitprocessing/blob/ad0bbdbe8445b115bf061d4682fea0cd02454197/.github/workflows/lint.yml#L36-L37)

> python3 -m pip install pyproject-deplister
> pyproject-deplister --extra dev --path pyproject.toml | grep ruff | sed 's/ //g' | xargs -I{} python3 -m pip install "{}"